integration/spark/src/main/scala/org/apache/spark/sql/CarbonDatasourceRelation.scala

```diff
@@ -55,18 +55,11 @@ class CarbonSource extends RelationProvider
   override def createRelation(
       sqlContext: SQLContext,
       parameters: Map[String, String]): BaseRelation = {
-    // if path is provided we can directly create Hadoop relation. \
-    // Otherwise create datasource relation
-    parameters.get("path") match {
-      case Some(path) => CarbonDatasourceHadoopRelation(sqlContext, Array(path), parameters, None)
-      case _ =>
-        val options = new CarbonOption(parameters)
-        val tableIdentifier = options.tableIdentifier.split("""\.""").toSeq
-        val identifier = tableIdentifier match {
-          case Seq(name) => TableIdentifier(name, None)
-          case Seq(db, name) => TableIdentifier(name, Some(db))
-        }
-        CarbonDatasourceRelation(identifier, None)(sqlContext)
+    val options = new CarbonOption(parameters)
+    if (sqlContext.isInstanceOf[CarbonContext]) {
```

Comment:
I don't think we can load the data using context type, earlier we are checking `path` parameter and then decides whether to use Hadoop relation or not. Because we could use dataframe.load(path). But now it can't work if we use carboncontext.load(path)